### PR TITLE
fix: hide empty non-Java folder ancestors

### LINKED ISSUE #1062: Empty src/main folder appears when Show Non-Java Resources is enabled
## Description

When **Show Non-Java Resources** is enabled, the Java Projects view can display an empty physical folder hierarchy such as `src/main` alongside the actual Java package roots (`src/main/java` and `src/main/resources`).

This was reported as part of #1044. This issue tracks only the empty `src/main` hierarchy, which is a rendering/data-model bug. The visibility of build output such as `target` is a separate filtering-policy question.

## Root cause

The project children returned by `java.getPackageData` combine Java classpath roots with `IJavaProject.getNonJavaResources()`.

The physical `src` folder can therefore be emitted as a non-Java folder even though its descendants are represented separately as Java package roots. When that folder is expanded, Java elements such as `src/main/java` and `src/main/resources` are filtered from its non-Java children, leaving an empty `src/main` hierarchy in the tree.

Relevant implementation:

- `jdtls.ext/.../PackageCommand.java#getProjectChildren`
- `jdtls.ext/.../parser/ResourceSet.java`
- `jdtls.ext/.../parser/JavaResourceVisitor.java`

## Expected behavior

Enabling **Show Non-Java Resources** should continue to show actual non-Java files, such as YAML and XML resources, but should not display ancestor folders that have no visible children because their content is already represented by Java package roots.

## Acceptance criteria

- Non-Java resource files remain visible.
- Empty duplicate ancestor hierarchies such as `src/main` are not shown.
- Existing Java source and resource package roots remain visible and functional.
- Add regression coverage for the returned project tree and an E2E UI scenario if practical.

Related: #1044

### PR DESCRIPTION
**Pull request by wenytang-ms**
## Summary

- hide physical non-Java folder ancestors when their entire subtree is already represented by Java model/package-root nodes
- keep real non-Java resources visible through their Java resource root
- add a Maven regression fixture and project-view test for the empty `src/main` hierarchy

This fixes the empty physical `src/main` tree reported when **Show Non-Java Resources** is enabled. It intentionally does not change the visibility policy for Maven `target` or other build-output folders.

## Root cause

`java.getPackageData` combines Java classpath roots with `IJavaProject.getNonJavaResources()`. A physical ancestor such as `src` could therefore be emitted as a normal folder even though `src/main/java` and `src/main/resources` were already represented as package roots. Expanding the physical folder filtered those Java elements and left an empty hierarchy.

## Testing

- JDTLS extension build and checkstyle
- `npm run compile`
- `npm run tslint`
- Maven Project View Tests: 15 passing
- AutoTest `java-dep-view-modes`: 36/36 passing with the fix
- Pre-fix A/B: 35/36 passing; only `verify-empty-physical-src-hidden` fails

Fixes #1062
Related to #1044

## UI verification

The Maven resource root and `application.yml` remain visible, while the duplicate physical `src` hierarchy is absent.

| Before | After |
| --- | --- |
| The extra physical `src` folder is shown below `.vscode`. | The extra `src` folder is removed; `target` follows `.vscode`. |
| Before: duplicate physical src folder | After: duplicate physical src folder removed |

UI verification for empty src hierarchy fix

### CODE REVIEW

File: jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/parser/ResourceSet.java [hunk 1]
```diff
@@ -152,4 +155,46 @@ private boolean shouldVisit(IResource resource) {
...
+                if (member instanceof IContainer && hasVisibleNonJavaResources((IContainer) member)) {
+                    return true;
+                }
+            }
+        } catch (CoreException e) {
+            JdtlsExtActivator.logException("Failed to inspect non-Java resources", e);
+            return true;
+        }
+        return false;
+    }
```
[the remark this answers is not in the document]
Comment (wenytang-ms) [reply jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/parser/ResourceSet.java:202]: Fixed in bdff6b1. Ordinary non-Java child folders are now treated as visible immediately; recursion only continues when the child is itself an ancestor of a source classpath entry. Added a regression test that creates an empty src/docs folder and verifies both src and docs remain visible.

**Review by chagong — APPROVED** [review]
